src/databricks/labs/ucx/hive_metastore/mapping.py

```diff
@@ -103,14 +125,90 @@ def skip_table(self, backend: StatementExecutionBackend, schema: str, table: str
         except BadRequest as br:
             logger.error(br)
 
-    def skip_schema(self, backend: StatementExecutionBackend, schema: str):
+    def skip_schema(self, schema: str):
         # Marks a schema to be skipped in the migration process by applying a table property
         try:
-            backend.execute(f"ALTER SCHEMA `{schema}` SET DBPROPERTIES('{self.UCX_SKIP_PROPERTY}' = true)")
+            self._backend.execute(f"ALTER SCHEMA `{schema}` SET DBPROPERTIES('{self.UCX_SKIP_PROPERTY}' = true)")
         except NotFound as nf:
             if "[SCHEMA_NOT_FOUND]" in str(nf):
                 logger.error(f"Failed to apply skip marker for Schema {schema}. Schema not found.")
             else:
                 logger.error(nf)
         except BadRequest as br:
             logger.error(br)
+
+    def _get_upgraded_tables(self) -> dict[str, TableInfo]:
+        upgraded_tables = {}
+        for catalog in self._ws.catalogs.list():
+            if not catalog.name:
+                continue
+            for schema in self._ws.schemas.list(catalog_name=catalog.name):
+                if not schema.name:
+                    continue
+                for table in self._ws.tables.list(catalog_name=catalog.name, schema_name=schema.name):
+                    if table.properties is not None and "upgraded_from" in table.properties:
+                        upgraded_tables[table.properties["upgraded_from"].lower()] = table
+        return upgraded_tables
```

Comment:
```suggestion
```
no longer needed because of `_is_target_exists`

Reply:
It is required. We are testing from both ends.
As we agreed upgraded_to cannot be trusted. We need to look at the upgraded_from as the source of truth.

Reply:
You are fetching table info in that method already